src/Microsoft.AspNetCore.Razor.Language/CodeGeneration/DefaultCodeRenderingContext.cs

```diff
@@ -81,6 +80,10 @@ internal class DefaultCodeRenderingContext : CodeRenderingContext
         // This will be initialized by the document writer when the context is 'live'.
         public IntermediateNodeVisitor Visitor { get; set; }
 
+        public override IEnumerable<IntermediateNode> Ancestors => _ancestors;
+
+        internal Stack<IntermediateNode> AncestorsInternal => _ancestors;
```

Comment:
That's not straightforward because we don't have IVT to the testcommon project and I don't want to add it.